onnxscript/ir/_tape.py

```diff
@@ -60,6 +65,13 @@ def op_multi_output(
 
         return node.outputs
 
+    def initializer(self, tensor: ir.TensorProtocol, name: str | None = None) -> ir.Value:
+        name = name or tensor.name
+        if name is None:
+            raise ValueError("Name must be provided for initializer.")
+        value = ir.Value(name=name, shape=tensor.shape, type=ir.TensorType(tensor.dtype), const_value=tensor)
```

Comment:
We can add some asserts (not None) here and there to appease the type checker and ensure robustness